src/Utils/request/query.ts

```diff
@@ -5,15 +5,26 @@ import { getResponseBody } from "@/Utils/request/request";
 import { QueryOptions, Route } from "@/Utils/request/types";
 import { makeHeaders, makeUrl } from "@/Utils/request/utils";
 
+// Extend the QueryOptions interface to include customHeaders
+export interface ExtendedQueryOptions<TBody> extends QueryOptions<TBody> {
+  customHeaders?: Record<string, string>;
+}
```

Comment:
The interface `QueryOptions` is defined exclusively for this itself. Feel free to move that interface here and directly add custom headers to that interface.

```suggestion

export interface ExtendedQueryOptions<TBody> extends QueryOptions<TBody> {
  headers?: HeadersInit;
}
```

Reply:
Made the changes as suggested